linalg/densemat.hpp

```diff
@@ -425,13 +425,20 @@ bool LinearSolve(DenseMatrix& A, double* X, double TOL = 1.e-9);
 /// Matrix matrix multiplication.  A = B * C.
 void Mult(const DenseMatrix &b, const DenseMatrix &c, DenseMatrix &a);
 
+/// Matrix matrix multiplication.  A = B^t * C.
+void MultTranspose(const DenseMatrix &b, const DenseMatrix &c, DenseMatrix &a);
```

Comment:
This functionality already exists. See 

https://github.com/mfem/mfem/blob/caf5d7b471571b2094ccdfd4b2ff772d76fc778c/linalg/densemat.hpp#L482-L483

Reply:
Hmm, I had not noticed, thanks. The interface is somewhat inconsistent in the naming, so I removed the implementation, but I would prefer to keep the function.

Reply:
I would prefer to have just the existing `MultAtb` and rename AddMultTranspose to `AddMultAtb`, I guess it's a matter of taste, though I would go with the most popular vote.

Reply:
There is a similar thing for sparse matrices `TransposeMult()`, so that is another option :smile: We should really vote about this.

Reply:
I agree with @psocratis, I think the names `MultAtB` and `AddMultAtB` are more precise and so a better choice here. Also, renaming the arguments to `A`, `B`, and `C` (in this order) will make things consistent with the function names.

Reply:
Ok, I wanted it to follow the terminology of the methods of `DenseMatrix`, but this also makes sense 😉 . Besides that we are missing `_a_` version here, so I will add also that one 👍 .